r/R/dataset.R

```diff
@@ -141,6 +141,42 @@ Dataset$create <- function(children, schema) {
 #' @export
 names.Dataset <- function(x) names(x$schema)
 
+#' @export
+dim.Dataset <- function(x) {
+
+  if (!inherits(x$format, "ParquetFileFormat")) {
+    stop(
+      "dim() is only currently implemented for Datasets that use parquet files.",
+      "Call collect() first to pull data into R.",
+      call. = FALSE
+    )
+  }
+
+  rows <- as.integer(sum(purrr::map_dbl(x$files, ~ParquetFileReader$create(.x)$num_rows)))
+  cols <- length(x$schema)
+
+  c(rows, cols)
+}
+
+#' @export
+dim.arrow_dplyr_query <- function(x) {
+
+  if (isTRUE(x$filtered)) {
+    rows <- nrow(x$.data)
+  } else {
+    warning("dim() is not fully implemented for arrow dplyr queries. ",
```

Comment:
Clarify that you're returning NA because there's a filter expression

Reply:
How about?
> Warning message:
dim() returns NA for the number of rows for arrow dplyr queries that call a filter expression. 
Call collect() first to pull data into R to access the number of rows.